### DISCUSSION TITLE
If ESIL emulation of an insn is incorrect, how to hotfix?

### DISCUSSION THREAD

**Discussion by jstaursky** [radareorg/radare2#23152] [Q&A] [answered]
This came up at work the other day, I was emulating some shellcode and noticed an instruction did not compute correctly. I don't recall the exact instruction and it might be fixed on the current version, but is there a way to hotfix esil instructions? Its not very easy to upgrade r2 in my workplace...

**Comment by trufae** [comment, marked as answer]
There are different ways:

* anal hints . Replace any instruction esil with (ahe command)
* Anal pins, execute a script when pc reaches a specific address
* create an arch plugin with the fixed esil and load it when you need it

Ideally updating r2 should be the way to go

**Reply by jstaursky** [reply]
Thanks! Could you give an example of doing it via anal hints?
